modules/openapi-generator/src/main/resources/ruby-client/faraday_api_client.mustache

```diff
@@ -0,0 +1,392 @@
+=begin
+{{> api_info}}
+=end
+
+require 'date'
+require 'faraday'
+require 'json'
+require 'logger'
+require 'tempfile'
+
+module {{moduleName}}
+  class ApiClient
+    # The Configuration object holding settings to be used in the API client.
+    attr_accessor :config
+
+    # Defines the headers to be used in HTTP requests of all API calls by default.
+    #
+    # @return [Hash]
+    attr_accessor :default_headers
+
+    # Initializes the ApiClient
+    # @option config [Configuration] Configuration for initializing the object, default to Configuration.default
+    def initialize(config = Configuration.default)
+      @config = config
+      @user_agent = "{{#httpUserAgent}}{{{.}}}{{/httpUserAgent}}{{^httpUserAgent}}OpenAPI-Generator/#{VERSION}/ruby{{/httpUserAgent}}"
+      @default_headers = {
+        'Content-Type' => 'application/json',
+        'User-Agent' => @user_agent
+      }
+    end
+
+    def self.default
+      @@default ||= ApiClient.new
+    end
+
+    # Call an API with given options.
+    #
+    # @return [Array<(Object, Integer, Hash)>] an array of 3 elements:
+    #   the data deserialized from response body (could be nil), response status code and response headers.
+    def call_api(http_method, path, opts = {})
```

Comment:
This template is very similar to api_client.mastache. The main difference is the implementation of call_api() and build_request(). Would it make sense to combine the two templates and simply pick the implementation of these methods based on the library selected?

Reply:
This file is ~400 lines.  Keeping both templates in sync when changes happen outside those methods will be a PITA I think. 
So I'd say for long term maintainability, it makes sense to merge them but I'm an outsider of this project, I hope maintainers can chime in.

Reply:
The generator supports a `library` option, which allows library specific overrides for individual files. Using this may make the logic and template separation clearer. The Java client generator is a good example. Let us know if you'd like to move toward that option.

Reply:
Yes, I think there's a library option added by this PR: 

https://github.com/OpenAPITools/openapi-generator/pull/3405/files#diff-23d62a3ed9c41293f930468ed3ee74a8R27

so the other 2 options (Ruby HTTP library names) are not needed

Reply:
> This template is very similar to api_client.mastache

I'll see if I can combine the two ...

Reply:
That makes sense. I'll remove it

> Yes, I think there's a library option added by this PR:
> 
> https://github.com/OpenAPITools/openapi-generator/pull/3405/files#diff-23d62a3ed9c41293f930468ed3ee74a8R27
> 
> so the other 2 options (Ruby HTTP library names) are not needed

Makes sense. I'll remove them.